server/src/main/java/org/candlepin/model/ProductContent.java

```diff
@@ -144,6 +144,22 @@ public int hashCode() {
     }
 
     /**
+     * Calculates and returns a version hash for this entity. This method operates much like the
+     * hashCode method, except that it is more accurate and should have fewer collisions.
+     *
+     * @return
+     *  a version hash for this entity
+     */
+    public int getEntityVersion() {
+        int hash = 17;
+
+        hash = 7 * hash + (this.content != null ? this.content.getEntityVersion() : 0);
```

Comment:
Do you think this might work?
```
        HashCodeBuilder hcb = new HashCodeBuilder(7, 17);
        hcb.append(enabled);
        if (this.content  != null)
            hcb.append(this.content.getEntityVersion());
```